worksheet.rb

```diff
@@ -31,4 +127,57 @@
 # - the total amount of money each driver has made
 # - the average rating for each driver
 # - Which driver made the most money?
-# - Which driver has the highest average rating?
\ No newline at end of file
+# - Which driver has the highest average rating?
+
+#~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#Q1 - the number of rides each driver has given
+
+driver_data.each do |driver_id, array_of_rides| #key is the driver_id, the value is an array of all the ride data
+  puts "Driver # #{driver_id}, has given #{array_of_rides.length} rides"
+end
+
+#~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#2  - the total amount of money each driver has made
+
+all_drivers_earnings = {} # Here create a hash for the totals for each driver to be used later to compare
+
+puts ""
+driver_data.each do |driver_id, array_with_ea_ride| #the value is an array
+    drivers_earnings = 0
+    array_with_ea_ride.each do |ea_ride_hash| #runs each on the array of hashes, so each ride {}
+          drivers_earnings += ea_ride_hash[:cost]
+    end
```

Comment:
You can simplify this using `sum`:

```suggestion
    drivers_earnings = array_with_ea_ride.sum { |ea_ride_hash| ea_ride_hash[:cost] }
```